plugins/settings-sync/src/com/intellij/settingsSync/SettingsSyncBridge.kt

```diff
@@ -77,6 +77,10 @@ class SettingsSyncBridge(
   internal fun initialize(initMode: InitMode) {
     coroutineScope.launch {
       try {
+        // Always explicitly flush settings – if this is not done before sending sync events, then remotely synced settings
+        // might not contain the most up–to–date settings state (e.g. sync settings will be stale).
+        saveIdeSettings()
```

Comment:
that might potentially lead to problems, such as https://youtrack.jetbrains.com/issue/IJPL-1245/NullPointerException-Cannot-invoke-com.intellij.openapi.application.impl.ReadMostlyRWLock.isReadAllowed-because-this.myLock-is (https://github.com/JetBrains/intellij-community/commit/2c938259247aaa1b6bc7a0c93b48e6c067192960)

Since, it happens inside the coroutine, I don't expect it to happen. But still a thing to keep in mind and watch

Reply:
Just to double check I understand — nothing needs to be done this, just a note for the future?

Reply:
@jaen yes